src/Gameplay/Pile.java

```diff
@@ -0,0 +1,51 @@
+package Gameplay;
+
+import Cards.Card;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.EmptyStackException;
+import java.util.Stack;
+
+public abstract class Pile extends JPanel {
+    //class is abstract because methods defined here are needed by many other classes
+    //such as the deck, the foundations, the waste pile and the stockpiles
+
+    protected Integer x, y;   //horizontal and vertical positions in the GUI
+    protected Stack<Card> cards; //a stack is better than a list because it is a LIFO data structure
```

Comment:
General recommendation : protected fields are evil, if you need derived class access, use protected manipulation functions.

Since x,y  and cards are pointers (not primitives), you only need getters.

```suggestion
    private Integer x, y;   //horizontal and vertical positions in the GUI
    private Stack<Card> cards; //a stack is better than a list because it is a LIFO data structure
    
    protected getX()
    /// and the rest
```